### PR TITLE
feat: Rubygem release engine

### LINKED ISSUE #491: Add a release engine for Rubygems
This is part of our overarching goal of supporting more package managers in Keygen: https://github.com/keygen-sh/keygen-api/issues/493.

See #679 for implementation of PyPI, and #753 for Tauri. Bulk of the PyPI logic is in [the controller](https://github.com/keygen-sh/keygen-api/blob/master/app/controllers/api/v1/release_engines/pypi/simple_controller.rb), with integration tests [here](https://github.com/keygen-sh/keygen-api/blob/master/features/api/v1/engines/pypi/simple/index.feature) and [here](https://github.com/keygen-sh/keygen-api/blob/master/features/api/v1/engines/pypi/simple/show.feature). The Tauri implementation can be found in the aforementioned PR, since it's much simpler than the PyPI PR, as the PR for PyPI contained a lot of ground work for future packaging systems and may not be particularly helpful.

Implementation-wise here, we need to implement the expected [API for Rubygems](https://github.com/KN4CK3R/gitea/blob/4b1f387cb90e15d0f1659b96df6a283f98c65438/routers/api/packages/api.go#L368-L378), in a way that's compatible with Keygen's existing distribution schema. May be able to use [the Rubygems lib](https://github.com/rubygems/rubygems) for heavy lifting here, but I'm not sure. Like PyPI and Tauri, the implementation should only be concerned with downloading gems, not uploading. Drafting releases and uploading artifacts to Keygen's distribution API can be done via [Keygen's CLI](https://keygen.sh/docs/cli/), or via the Dashboard.

For an overview on the API endpoints expected by Rubygems, check out this blog post: https://keygen.sh/blog/how-to-license-and-distribute-a-private-rubygem/. We can use that to model the Rubygems engine here. This engine is meant to replace the workflow in that blog post — it should be a turn-key way to distribute gems.

In the end, I want a customer to be able to upload a `.gem` file (via `gem build`) to the distribution API we already have, and allow their end-users to `gem install` that gem like they would from https://rubygems.org.

Implementation should include [integration tests](https://github.com/keygen-sh/keygen-api/blob/master/features/api/v1/engines/tauri/v2/upgrade.feature). lmk of any questions.

### PR DESCRIPTION
**Pull request by mehulmathur16**
/closes #491
/claim #491

### CODE REVIEW

File: app/controllers/api/v1/release_engines/rubygems/simple_controller.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                  .where(release_filetypes: { key: %w[gem gz zip] })
+                  .order(created_at: :desc)
+                  .first
+      authorize! artifact
+
+      # Render JSON response
+      render json: {
+        url: vanity_v1_account_release_artifact_url(artifact.account, artifact, filename: artifact.filename),
+        version: release.version,
+        pub_date: release.created_at.rfc3339(3),
```
Comment (ezekg) [inline app/controllers/api/v1/release_engines/rubygems/simple_controller.rb:41]: This is the required format for Tauri, not Rubygems. This does not follow what Rubygems expects, i.e. this won't be installable via `gem install foo --source https://rubygems.pkg.keygen.sh/bar`.

**Review by ezekg — CHANGES_REQUESTED** [review]

**Comment by ezekg** [comment]
This isn't really what the issue is describing. Are you sure you understand what's required?

**Comment by mehulmathur16** [comment]
@ezekg I'll review the details in issue #491 again to ensure I'm fully aligned with the requirements.

**Comment by ezekg** [comment]
There's implementations like [Gitea's](https://github.com/KN4CK3R/gitea/blob/4b1f387cb90e15d0f1659b96df6a283f98c65438/routers/api/packages/rubygems/rubygems.go), [GitLab's](https://github.com/gitlabhq/gitlabhq/tree/2b42932ce5871c9c240aa9a56745cec8807c3876/app/services/packages/rubygems), and [Rubygems.org](https://github.com/rubygems/rubygems.org) to study for a general idea of what I'm thinking here. In the end, I want a user to be able to upload a `.gem` file, and essentially automate all of the manual work in [this blog post](https://keygen.sh/blog/how-to-license-and-distribute-a-private-rubygem/) — ultimately allowing a user to `gem install` the package like a normal gem from https://rubygems.org.

**Comment by ezekg** [comment]
Closing due to inactivity. Feel free to open a new PR if you pick this up again. 👍

**Pull request closed without merging**